### ISSUE TITLE
Adding 775 redirects causes local development server to fail with memory error

### ISSUE THREAD

**Issue by attaboy** [netlify/cli#709]
**- Do you want to request a _feature_ or report a _bug_?**
Bug

**- What is the current behavior?**
I have a `netlify.toml` file with about 775 redirect rules in it. (I have a blog that used to be PHP and I want to redirect to .html — this is to work around the lack of support for filename wildcards.) I'm using jekyll to build the site.

This works fine in the production environment, but with the `netlify dev` command, the local dev server usually (though not always) fails to start up because of a memory error in `netlify-redirector`:

```
Cannot enlarge memory arrays. Either (1) compile with  -s TOTAL_MEMORY=X  with X higher than the current value 16777216, (2) compile with  -s ALLOW_MEMORY_GROWTH=1  which allows increasing the size at runtime, or (3) if you want malloc to return NULL (0) instead of this abort, compile with  -s ABORTING_MALLOC=0 
Cannot enlarge memory arrays. Either (1) compile with  -s TOTAL_MEMORY=X  with X higher than the current value 16777216, (2) compile with  -s ALLOW_MEMORY_GROWTH=1  which allows increasing the size at runtime, or (3) if you want malloc to return NULL (0) instead of this abort, compile with  -s ABORTING_MALLOC=0 
abort("Cannot enlarge memory arrays. Either (1) compile with  -s TOTAL_MEMORY=X  with X higher than the current value 16777216, (2) compile with  -s ALLOW_MEMORY_GROWTH=1  which allows increasing the size at runtime, or (3) if you want malloc to return NULL (0) instead of this abort, compile with  -s ABORTING_MALLOC=0 ") at Error
    at jsStackTrace (/Users/luke/.nvm/versions/node/v11.14.0/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.js:1070:13)
    at stackTrace (/Users/luke/.nvm/versions/node/v11.14.0/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.js:1087:12)
    at abort (/Users/luke/.nvm/versions/node/v11.14.0/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.js:8415:44)
    at abortOnCannotGrowMemory (/Users/luke/.nvm/versions/node/v11.14.0/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.js:1170:3)
    at enlargeMemory (/Users/luke/.nvm/versions/node/v11.14.0/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.js:1175:3)
    at wasm-function[3745]:65
    at wasm-function[2919]:7190
    at wasm-function[3461]:32
    at wasm-function[2431]:289
    at wasm-function[2426]:1164
```

**- If the current behavior is a bug, please provide the steps to reproduce.**
Add 775 (?) `[[Redirects]]` to the `netlify.toml` file and then run the `netlify dev` CLI command. It could also be related to using jekyll 4.0.0 to build the site, but it works fine when there are no redirects or very few redirects.

**- What is the expected behavior?**
The server should start up normally and redirect paths as requested.

**- Local Environment Information**
```
  System:
    OS: macOS 10.15.3
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  Binaries:
    Node: 11.14.0 - ~/.nvm/versions/node/v11.14.0/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.7 - ~/.nvm/versions/node/v11.14.0/bin/npm
  Browsers:
    Chrome: 80.0.3987.106
    Firefox: 72.0.2
    Safari: 13.0.5
```

**Comment by mraerino** [comment]
It would be great if you could show an example of such a redirect.
Did you know that Netlify allows you to access any html file without its file extension?

**Comment by attaboy** [comment]
> It would be great if you could show an example of such a redirect.

The redirects all look like this:

```
[[redirects]]
  from: “/archives/2012/12/000123.php”
  to: “/archives/2012/12/000123.html”
```

If there was a way to write regular expressions or wildcards with extensions,  I’d do that, but AFAICT doing it one at a time is the only way to do what I want. 

> Did you know that Netlify allows you to access any html file without its file extension?

Ah so perhaps I should redirect to remove the extension! I did not know that. However the problem I’m solving here is that I’m migrating a PHP site where all the public URLs used to end in `.php`. I want to redirect all those requests so old URLs and search results continue to work.

**Comment by mraerino** [comment]
maybe you want to call your new files like `/archives/2012/12/000123.php.html` until we figured this out?

**Comment by attaboy** [comment]
> maybe you want to call your new files like `/archives/2012/12/000123.php.html` until we figured this out?

Heh, I did not consider that idea. Clever!

My production site is working just fine with all the redirects in place, so I'll leave it for now. I can probably just disable the redirects when I want to test the site locally.

**Comment by RaeesBhatti** [comment]
Thank you for reporting this @attaboy . We've identified the cause of this problem. A fix for it is in the works https://github.com/netlify/libredirect/pull/88

**Comment by RaeesBhatti** [comment]
@attaboy A fix for this has been released in `netlify-cli@2.42.0`. Please update and let us know if it is still an issue.
